### PR TITLE
add enum support

### LINKED ISSUE #289: Add support for creating enums
It seems there is no support right now for creating enums. (If i didn't miss anything?)

### PR DESCRIPTION
**Pull request by TarekkMA**
closes #289

### CODE REVIEW

**Comment by TarekkMA** [comment]
@matanlurey can you take a look as the PR

File: example/example.dart [hunk 1]
```diff
@@ -3,13 +3,15 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'package:code_builder/code_builder.dart';
+import 'package:code_builder/src/specs/enum.dart';
```
Comment (natebosch) [inline example/example.dart:6]: The examples shouldn't need to import from `src`. Is there something else that needs to be exported?

File: example/example.dart [hunk 2]
```diff
@@ -52,3 +54,32 @@ String scopedLibrary() {
...
+///  @JsonKey('m')
+///  Metric,
+///  @JsonKey('i')
+///  Imperial
+/// }
+/// ```
+String jsonEnum() {
+  final values = <EnumValue>[
+    EnumValue((b) => b
+      ..name = 'Metric'
```
Comment (natebosch) [inline example/example.dart:71]: This should be `'metric'` in our examples.

File: lib/src/specs/enum.dart [hunk 3]
```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
```
Comment (natebosch) [inline lib/src/specs/enum.dart:1]: [nit] 2020

Comment (TarekkMA) [reply lib/src/specs/enum.dart]: Okay, I have updated the copyright date to 2020

File: example/example.dart (hunk 2 above)
Comment (TarekkMA) [reply example/example.dart:70]: What do you mean? Should I rename the enum values to be lowercase?

Is that what should it look like?
```dart
enum Unit {
 @JsonKey('m')
 metric,
 @JsonKey('i')
 imperial
}
```

File: example/example.dart (hunk 1 above)
Comment (TarekkMA) [reply example/example.dart]: okay this import was useless.

File: example/example.dart (hunk 2 above)
Comment (natebosch) [reply example/example.dart:70]: Yes. We always follow best practices from Effective Dart in our examples.

https://dart.dev/guides/language/effective-dart/style#do-name-other-identifiers-using-lowercamelcase

Comment (TarekkMA) [reply example/example.dart]: Oh, that is right, it should be fixed now.

**Review by natebosch — APPROVED** [review]

**Comment by natebosch** [comment]
This looks like a great feature. Thanks!

We'll need some internal validation before publishing.

**Pull request merged**